### PR TITLE
Added a user's activity to their profile page

### PR DESCRIPTION
**Pull request by aylinev2**
- I added an area on a user's profile page that shows all the messages that the user has sent

### CODE REVIEW

**Review by acolwell — COMMENTED** [review]
This looks like it would work, but I have some questions to help guide you toward something a little better. :)

1. What is the worst case runtime complexity for these changes? How could you improve on that?
2. What patterns do other parts of the code base use for filtering messages?
3. How would you refactor this code to make it more testable?

**Comment by aylinev2** [comment]
- I added a hashmap that contains a list of a specific user's messages
- When the server is started up the setMessages() function is called so I thought it would be best to just sort the messages by author once right when the new list of messages is set
- I also added some code underneath addMessage() so that the sortMessagesByAuthor() function doesn't have to be called again each time theres a new message in the session

File: src/main/java/codeu/model/store/basic/MessageStore.java [hunk 1]
```diff
@@ -70,6 +75,23 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
     return messages;
   }
 
+  /** Maps each message to its author. */
+  private void sortMessagesByAuthor(){
```
Comment (acolwell) [inline src/main/java/codeu/model/store/basic/MessageStore.java]: "sort" -> "split" or "sort" -> "filter" ? The main reason I'm suggesting this is "sort" implies to me that it is taking an message list and creating another list that is sorted by author. That isn't quite what you are doing here. Another option might be renaming this to something like "populateUserMessagesMap".

File: src/main/java/codeu/model/store/basic/MessageStore.java [hunk 2]
```diff
@@ -70,6 +75,23 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
...
+    for (Message message : messages){
```
Comment (acolwell) [inline src/main/java/codeu/model/store/basic/MessageStore.java]: Replace userMessages with an empty map so this doesn't end up containing old messages if setMessages() is called more than once.

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 3]
```diff
@@ -48,6 +71,24 @@ TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("k
...
      <% } %>
 
+     <hr/>
+     <h2><%= user.getName()%>'s Sent Messages</h2>
+     
+     <div id="activity">
+      <ul>
+     <%
+      for (Message message : messages) {
+        String author = UserStore.getInstance().getUser(message.getAuthorId()).getName();
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/view/profile.jsp]: I think you can avoid this lookup. Isn't message.getAuthorId() always equivalent to user.getId()? If so, I think you could just use user.getName() here instead.

Also consider moving the messages declaration to right before the loop. I think that would make it easier to see that the variable contains only messages for user.getId() and would add extra confidence that user.getName() is appropriate here.

**Review by acolwell — APPROVED** [review]
Looking pretty good to me. Just a few comments.

Also I believe if you add  "Fixes Issue #19 " to the merge message this pull request will automagically get associated with the issue.

**Pull request merged**